python-for-android/dists/kolibri/src/main/java/org/kivy/android/PythonUtil.java

```diff
@@ -123,16 +124,30 @@ protected static void recursiveDelete(File f) {
         f.delete();
     }
 
+    public static String getPrivateVersion() {
+        // We read this directly from the VERSION_CODE,
+        // so that any upgrade of the app causes the Python
+        // code to be extracted again.
+        return Integer.toString(BuildConfig.VERSION_CODE);
+    }
```

Comment:
Could this just be a static var on the class?

Reply:
Oh, yes, probably.

Reply:
Updated - still seems to work as intended.